tokio/tests/fs_copy.rs

```diff
@@ -1,5 +1,5 @@
 #![warn(rust_2018_idioms)]
-#![cfg(all(feature = "full", not(target_os = "wasi")))]
+#![cfg(all(feature = "full", not(target_os = "wasi")))] // Wasi does not support file operations
```

Comment:
Well, technically WASI *does* support file operations. It's just, that it needs a preopened directory as the base directory. For the tests one could define a root directory FD preopened by wasmtime.

But, the big point is, that tokio uses threads for all blocking filesystem operations.

Reply:
If the WASI file operations are async-compatible, then we certainly could use them for `tokio::fs` in the future. However, I don't think that needs to happen in this PR.

Reply:
I'll have a follow-up to address any remaining concerns.